src/app/modules/react-migration/react-migration.controller.ts

```diff
@@ -38,7 +43,67 @@ export const ADMIN_COOKIE_OPTIONS_WITH_EXPIRY = {
 
 const logger = createLoggerWithLabel(module)
 
-const serveFormReact: ControllerHandler = (_req, res) => {
+const serveFormReact =
+  (isPublicForm: boolean): ControllerHandler =>
+  (req, res, next) => {
+    const reactFrontendPath = path.resolve('dist/frontend')
+    logger.info({
+      message: 'serveFormReact',
+      meta: {
+        action: 'routeReact.serveFormReact',
+        __dirname,
+        cwd: process.cwd(),
+        reactFrontendPath,
+      },
+    })
+
+    if (isPublicForm && !!get(req.params, 'formId')) {
+      return servePublicFormReact(
+        req as Request<
+          { formId: string },
+          unknown,
+          unknown,
+          Record<string, string>
+        >,
+        res,
+        next,
+      )
+    }
+
+    let reactHtml = readFileSync(path.join(reactFrontendPath, 'index.html'), {
```

Comment:
⚠️ We should not do `readFileSync()` in code flows triggered from user inputs!! ⚠️ 

Use the async `fs.readFile()` instead and if you want the promise version of it, transform it with `Util.promisy()`